lib/src/main/java/com/dynatrace/metric/util/Normalize.java

```diff
@@ -136,7 +150,39 @@ static String dimensionValue(String value) {
     return value;
   }
 
+  /**
+   * Fast check: Does the dimension value need to be escaped at all?
+   *
+   * @param dimensionValue The dimension value
+   * @return True if it contains characters that need to be escaped according to the Dynatrace
+   *     metrics ingestion protocol, false otherwise.
+   */
+  static boolean needToEscapeDimensionValue(String dimensionValue) {
+    int len = dimensionValue.length();
+    if (len > dv_max_length) {
+      return true;
+    }
+
+    for (int i = 0; i < len; i++) {
+      char c = dimensionValue.charAt(i);
+      if (c == dv_characterToEscape_quote
+          || c == dv_characterToEscape_equals
+          || c == dv_characterToEscape_blank
+          || c == dv_characterToEscape_comma
+          || c == dv_characterToEscape_backslash) {
+        return true;
+      }
+    }
+
+    return false;
+  }
+
   static String escapeDimensionValue(String val) {
+    // Fast pass: Only escape if escaping is actually necessary
```

Comment:
I just realized that a null check is missing here, and dimension value might be null (if somebody passed null in the creation)

Reply:
I can do that in a follow-up PR, but i think this might be important to guard against.

Reply:
Added null/empty check on given dimension.